sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultKeyStore.java

```diff
@@ -0,0 +1,325 @@
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.jca;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.KeyStoreSpi;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableEntryException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.logging.Logger;
+
+import static java.util.logging.Level.INFO;
+import static java.util.logging.Level.WARNING;
+
+/**
+ * The Azure KeyVault implementation of the KeyStoreSpi.
+ */
+public class KeyVaultKeyStore extends KeyStoreSpi {
+
+    /**
+     * Stores the logger.
+     */
+    private static final Logger LOGGER = Logger.getLogger(KeyVaultKeyStore.class.getName());
+
+    /**
+     * Stores the list of aliases.
+     */
+    private List<String> aliases;
+
+    /**
+     * Stores the certificates by alias.
+     */
+    private final HashMap<String, Certificate> certificates = new HashMap<>();
+
+    /**
+     * Stores the certificate keys by alias.
+     */
+    private final HashMap<String, Key> certificateKeys = new HashMap<>();
+
+    /**
+     * Stores the creation date.
+     */
+    private final Date creationDate;
+
+    /**
+     * Stores the key vault client.
+     */
+    private KeyVaultClient keyVault;
+
+    /**
+     * Constructor.
+     *
+     * <p>
+     * The constructor uses System.getProperty for
+     * <code>azure.keyvault.uri</code>, <code>azure.keyvault.tenantId</code>,
+     * <code>azure.keyvault.clientId</code>,
+     * <code>azure.keyvault.clientSecret</code> to initialize the keyvault
+     * client.
+     * </p>
+     */
+    public KeyVaultKeyStore() {
+        creationDate = new Date();
+        String keyVaultUri = System.getProperty("azure.keyvault.uri");
+        String tenantId = System.getProperty("azure.keyvault.tenantId");
+        String clientId = System.getProperty("azure.keyvault.clientId");
+        String clientSecret = System.getProperty("azure.keyvault.clientSecret");
+        keyVault = new KeyVaultClient(keyVaultUri, tenantId, clientId, clientSecret);
+    }
+
+    @Override
+    public Enumeration<String> engineAliases() {
+        if (aliases == null) {
+            aliases = keyVault.getAliases();
+        }
+        return Collections.enumeration(aliases);
+    }
+
+    @Override
+    public boolean engineContainsAlias(String alias) {
```

Comment:
Consider adding code snippets in Javadocs for public API.

Reply:
This is an API that has been part of Java for a long time and as such has its own documentation so no need to add code snippets here.